storage/src/main/java/org/apache/kafka/server/log/remote/storage/RemoteLogManagerConfig.java

```diff
@@ -356,6 +362,29 @@ public static ConfigDef configDef() {
 
     public RemoteLogManagerConfig(Map<?, ?> props) {
         super(configDef(), props);
+        this.remoteLogIndexFileCacheTotalSizeBytes = getLong(REMOTE_LOG_INDEX_FILE_CACHE_TOTAL_SIZE_BYTES_PROP);
+        this.remoteLogManagerCopyMaxBytesPerSecond = getLong(REMOTE_LOG_MANAGER_COPY_MAX_BYTES_PER_SECOND_PROP);
+        this.remoteLogManagerFetchMaxBytesPerSecond = getLong(REMOTE_LOG_MANAGER_FETCH_MAX_BYTES_PER_SECOND_PROP);
+        this.remoteFetchMaxWaitMs = getInt(REMOTE_FETCH_MAX_WAIT_MS_PROP);
+    }
+
+    /**
+     * Update the current configuration with the given properties.
+     * @param props
+     */
+    public void updateCurrentConfig(Map<?, ?> props) {
+        if (props.containsKey(REMOTE_LOG_INDEX_FILE_CACHE_TOTAL_SIZE_BYTES_PROP)) {
```

Comment:
It seems to me this way has following disadvantages.

1. it is not thread-safe
2. those configs must be equal to `DynamicRemoteLogConfig#ReconfigurableConfigs`, and it could be easy to forget to make them consistency
3. there are two sources of those configs - `props` and inner variable - that is a bit chaos

If we want to make all "created" `RemoteLogManagerConfig` can see "latest" configs and fix above disadvantages, a simple solution is `RemoteLogManagerConfig` can keep a reference of `KafkaConfig` instead of having KafkaConfig snapshot.  

WDYT?

Reply:
Agreed with @chia7712 , 
we should keep the consistency along with thread-safty.

Reply:
We cannot add the 'core' module dependency to the 'storage' module. So, `kafka.core.KafkaConfig` can't be referenced in RemoteLogManagerConfig.

Reply:
> 3. there are two sources of those configs - props and inner variable - that is a bit chaos

One option is to follow the approach that you have suggested in [this](https://github.com/apache/kafka/pull/16199#discussion_r1627934940) comment to avoid exposing the unrelated/unexpected methods.

Reply:
> We cannot add the 'core' module dependency to the 'storage' module. So, kafka.core.KafkaConfig can't be referenced in RemoteLogManagerConfig.

My point was to use  `KafkaConfig` to create `RemoteLogManagerConfig`, and so `RemoteLogManagerConfig` will have a reference to `AbstractConfig` ( and its point is `KafkaConfig`). For example:
```java
public RemoteLogManagerConfig(AbstractConfig configs)
```

Also, my another concern is  - how we handle other similar configs in the future? add all of them to `updateCurrentConfig` (it means they have two kinds of configs sources - props and individual dynamic volatile fields) ? or make them reuse a class which can reflect the latest configs?

Reply:
I was under the assumption that when we refer to `AbstractConfig`, then the `get` method will always hit the `AbstractConfig.get` instead of the overriden `KafkaConfig.get` method (which uses the current config). Updated the patch. Thanks for the review!

Reply:
Small suggestions `KafkaConfig` in server is now inheriting `AbstractKafkaConfig` and am already working on moving most getters into `AbstractKafkaConfig` here #16307 so maybe we can refer to `AbstractKafkaConfig` instead of `AbstractConfig` and any getters you need define it in `AbstractKafkaConfig`.  Specially that `AbstractKafkaConfig` is going to replace `KafkaConfig` at some point`

Reply:
@OmniaGM that will go back to face the issue - `AbstractKafkaConfig` is in server module :(

Reply:
>  any getters you need define it in AbstractKafkaConfig

that is another design we need to discuss. If we will have different config class (for example:`RemoteLogManagerConfig`) for each component, do we need to add all getters to `KafkaConfig`/`AbstractKafkaConfig`? Or we can follow the pattern of `RemoteLogManagerConfig` that `KafkaConfig` can create every config class to get related config? That is a trade-off, and currently I prefer the later as it avoids a huge `KafkaConfig` and `KafkaConfig` is still a center that we can get all configs from it.

Reply:
> @OmniaGM that will go back to face the issue - AbstractKafkaConfig is in server module :(

:/ yeah this is getting more complicated. I just feel using a very generic `AbstractConfig` instead of `KafkaConfig` make the code bit unreadable as you need to really to follow what config is passed. 

> any getters you need define it in AbstractKafkaConfig

We don't need to keep the getters there for ever however until we move everything out we need a place for `KafkaConfig` that not core as this is the hard no module to depend on in all the other modules.

Reply:
> :/ yeah this is getting more complicated. I just feel using a very generic AbstractConfig instead of KafkaConfig make the code bit unreadable as you need to really to follow what config is passed.

agree as we leverage `KafkaConfig` to see "latest" configs, and that is something behind the `AbstractConfig`. 

> We don't need to keep the getters there for ever however until we move everything out we need a place for KafkaConfig that not core as this is the hard no module to depend on in all the other modules.

it seems eventually we need a module to run server with all required code, and so I assume `server` module is the one (but it will get smaller than core module).

Also, `AbstractKafkaConfig` does a great job that it does not declare all `ConfigDef` by itself. Instead, it collect all `ConfigDef` from each config class. That is why I feel we can follow the pattern to deal with other config classes.